### PR TITLE
v1.1.0 — Per-position Stock Research page (closes #280)

### LINKED ISSUE #280: Add per-position Stock Research page routed from dashboard Review → CTA
## Motivation

The Positions table's per-row **Review →** CTA in the Next Action column currently routes to `/journal?position={id}` — the trade ledger — which surfaces history, not insight. This is a user-reported UX gap: *"I want to learn about the stock in the review, not data on CCs."*

Two focused tactical destinations already exist:
- `/positions/{id}/recovery` — recovery engine (#182, shipped)
- `/options?ticker={X}` — CC scanner (used by [P2] action card)

This issue adds the **research** destination: learn about the company you hold. It also **supersedes any per-row Review-link routing fix** — do not file that separately.

---

## Scope — Six sections, in render order

| # | Section | Purpose |
|---|---|---|
| **A** | Business snapshot | ~30s paragraph: what the company does, key segments, market cap. |
| **B** | Price chart w/ your basis + trade markers | Price line over a configurable window. Horizontal lines for broker basis and adjusted basis. Annotations for user's own trade events pulled from `Position.trades`. |
| **C** | Why did this move? | Earnings-date annotations (we already cache via `get_cached_next_earnings_date`). News-event markers via Alpha Vantage NEWS_SENTIMENT — **open question for designer**. |
| **D** | Financial scorecard | Revenue, EPS, gross/operating margins, last 8–12 quarters, trend lines. Leading source: Alpha Vantage `INCOME_STATEMENT`; yfinance fallback. **Source decision flagged for CTO plan phase.** |
| **E** | Regression: what drives this stock? | Multi-factor regression of ticker returns against a preset basket (SPY + sector ETF + 10Y yield). Output: "X% market, Y% rates, Z% idiosyncratic." This surfaces the app's namesake feature as one focused section — not the existing `/analysis` config-everything experience. Designer to decide: fixed preset vs. inline configurability. |
| **F** | Your thesis (freeform note, persistent per-position) | Plain-text note scoped to position id: why you bought it, what would change your mind. **Storage open question for CTO plan:** new column on `Position` vs. separate `position_notes` table. |

**Page header (nav only, not content):** three link-outs —
- `Recovery plan →` (`/positions/{id}/recovery`)
- `Scan covered calls →` (`/options?ticker={X}`)
- `Trade ledger →` (`/journal?position={id}`)

---

## Out of scope

- No IV, realized-vol, CC scanner data, or options chain — those live on their own pages.
- No analyst buy/sell ratings or sentiment scores in isolation.
- No news headline feed (chart event markers only).
- No multi-position comparison (one position at a time).

---

## Design decisions for the designer to flag/resolve

1. **Route:** `/positions/{id}/review` (recommended — position-scoped, basis + trades available natively) vs. `/research/{ticker}` (ticker-scoped, needs a position-id query param for overlays).
2. **Section C:** include Alpha Vantage NEWS_SENTIMENT event markers, or skip?
3. **Section E:** fixed preset basket vs. inline configurability?
4. **Section F storage:** `Position` column vs. `position_notes` table?

---

## Acceptance criteria

- [ ] Clicking **Review →** on any open position in the dashboard Positions table routes to the new research page for that position (not to `/journal`).
- [ ] Sections A–F render in the order specified above.
- [ ] Page skeleton renders within 300 ms; data hydrates section-by-section as it arrives.
- [ ] If any data source fails (Alpha Vantage rate-limit, yfinance hiccup), the affected section renders an explicit "data unavailable, retry" tile — the page does not 500.
- [ ] Freeform thesis note (Section F) persists across reloads, scoped to position id.
- [ ] All three header link-outs are present and route correctly.
- [ ] **Automated test coverage (required before merge):**
  - [ ] Backend: pytest in `backend/tests/` — thesis note round-trips, missing-data fallback responses.
  - [ ] Frontend e2e: Playwright in `frontend/e2e/` — page load, basis lines render at correct y-values, thesis note round-trip, missing-data fallback tile, header link-outs.

---

## SDLC gates required

- **Phase 1.5 (designer)** — mandatory per CLAUDE.md; new rendered UI screen.
- **Phase 2 (CTO plan)** — required before implementation; new data sources + storage decisions to settle.

---

## Related

- **Supersedes** any standalone per-row Review-link routing fix (do not file separately).
- **Related: #182** — Recovery Plan engine (shipped); the recovery card stays as-is, this page is adjacent.
- **Related: #244** — BTC detail screen (shipped); same precedent of a position-scoped detail screen with a focused purpose.
- **Possibly related:** existing `/analysis` page — it's regression-deep-dive; this page surfaces a *preset* regression as one section of six.

---

## Size

`size: xl` — new page with six data-fetching sections, two backend open questions, one new route, chart rendering, and a note persistence feature. Should be decomposed into sub-issues during the CTO plan phase.

### PR DESCRIPTION
**Pull request by ssandy33**
## What this release is

**v1.1.0 — MINOR bump.** New per-position research surface — answers "what do I own?" rather than "what trades did I make?" — replaces the dashboard's per-row `Review →` CTA destination from the trade journal to a purpose-built research page.

Closes: **#280**
PRD: [Discussion #282](https://github.com/ssandy33/regress/discussions/282)
Milestone: TBD (CPO assigns at release time)

## Design lineage

- **#280** — feature issue
- **PRD #282** — full product spec (Discussion, per memory convention)
- **Designer spec + mock** (untracked, local)
- **CTO plan** at `backend/design-specs/issue-280-stock-research-plan.md` (untracked) — 5-worker parallel DAG

## The page (`/positions/{id}/review`)

Six sections in render order, pure research, no tactical garnish:

| # | Section | Purpose | Source |
|---|---|---|---|
| **A** | Business snapshot | What the company does, segments, market cap | yfinance `Ticker.info` |
| **B** | Price chart w/ your basis + trade markers | Where you are relative to break-even, visualized | existing `/api/data` + Position.trades |
| **C** | Why did this move? | Earnings annotations on the same chart | AV `EARNINGS` (cached) |
| **D** | Financial scorecard | Revenue / EPS / margins last 8 quarters | AV `INCOME_STATEMENT`, yfinance fallback |
| **E** | Regression decomposition | Multi-factor: SPY + sector ETF + DGS10. Plain-English summary headline + donut + beta table. | existing regression engine + FRED |
| **F** | Your thesis (freeform note, persistent) | Why you bought, what would change your mind | new `position_notes` table |

Header link-outs (navigation only, not content): `Recovery plan →`, `Scan covered calls →`, `Trade ledger →`.

## Parallel-execution worker DAG

```
feat/v1.1.0 (off main)
├── W-A  feat/v1.1.0-w-a-business-financials   (Section A + D services + yfinance + sector map)
├── W-B  feat/v1.1.0-w-b-price-history         (Section B + C endpoint + AV historical earnings)
├── W-D  feat/v1.1.0-w-d-regression            (Section E endpoint + summary rules)
├── W-F  feat/v1.1.0-w-f-thesis-notes          (Section F endpoint + new position_notes table)
└── W    feat/v1.1.0-w-bridge                  (frontend + 2 missing endpoint shells + Review CTA)
```

Merge order: W-F → W-A → W-B → W-D → W. Two text-conflicts during W-B and W-D merges (`research.py`, `alpha_vantage_client.py`, `sector_etf_map.py`, `test_sector_etf_map.py`) — all resolved by superset/union, no semantic loss.

## What changed

- **5 new backend endpoints** under `/api/positions/{id}/research/{business|price-history|financials|regression|thesis}` (GET + PUT for thesis).
- **New `position_notes` table** for thesis persistence (single-row, version-counter; idempotent migration via existing `Base.metadata.create_all` pattern).
- **New `yfinance>=0.2.50` backend dep** (first yfinance dep in the codebase — explicitly authorized by maintainer; full sub-deps list documented in W-A's worker report).
- **Sector → ETF map** module with 11 GICS sectors → SPDR ETFs (XLF, XLK, XLV, XLE, XLP, XLY, XLI, XLB, XLU, XLRE, XLC).
- **Plain-English regression summary** generator implementing design spec §4 rules verbatim (backend-generated, NOT frontend-rules, per locked decision).
- **AV client extensions** (append-only): `get_income_statement`, `get_historical_earnings`, `_filter_since`, new caches.
- **13 new frontend files**: route shell, top-level page, 6 section components + header + caveats footer + trade-marker vocab + hook + Playwright e2e.
- **Dashboard Review CTA rewired**: `nextActionCta()` now routes `Review` → `/positions/{id}/review` (was `/journal?position={id}`).
- **3 superseded acceptance tests** in `test_acceptance_phase4.py` marked skip with rationale — they enforced a blanket yfinance ban that the PRD explicitly authorizes overriding for the research surface.

## Test plan

- [x] Backend: **1438 passed, 13 skipped** on integrated tip (full suite, includes 120+ new tests across the 4 backend workers)
- [x] Frontend: ESLint clean (0 errors), `next build` succeeds
- [x] Frontend smoke: **10/10 Playwright tests pass** in `stock-research.spec.js`
- [x] Dashboard regression check: existing `dashboard-positions-triage` and `dashboard-rule-monitor` specs still pass
- [ ] CI on integration branch (this PR) — will validate
- [ ] Post-merge: verify in prod by clicking Review on a real position

## Locked design decisions (PRD-resolved, not re-litigated)

1. Route: `/positions/{id}/review` (position-scoped)
2. Section C news markers: **deferred to v2** (earnings only in v1)
3. Section E regression basket: **fixed preset** (SPY + sector ETF + DGS10)
4. Section F storage: **separate `position_notes` table** with version counter

Plus 6 designer-flagged minor decisions (window picker hidden in v1, 8 quarters in Section D, backend-generated summary, 4000-char thesis cap, non-sticky header, no logo in Section A) all baked in.

## Out of scope (intentional, follow-ons)

- News-event markers (Alpha Vantage NEWS_SENTIMENT) — separate v1.2/v1.1.x follow-on
- Window picker UI — endpoints accept `window` param but UI is fixed at 1Y
- Inline regression basket configurability — `/analysis` remains the deep-dive
- Vector container-log shipping (sibling to v1.0.9 Axiom logging) — separate
- Thesis history UI (version field exists, but only the latest is rendered in v1)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by ssandy33 — COMMENTED** [review]
## Verdict: Approve

**Counts:** Blocking 0 · Should-fix 2 · Nits 4

Holistic Phase-5 code review of the v1.1.0 Stock Research surface (5-worker fan-out, ~3,000 LOC). I read the router, every new service module, every changed schema, the hook, every section component, the Playwright suite, and a representative slice of the 103 new backend tests. CI was running while I reviewed; I did not block on it.

The code is in great shape. The PR follows project conventions tightly (cache-first read paths, sanitized 5xx details, per-section failure isolation, frozen test IDs end-to-end, render-phase hydration for React 19). No security or test-coverage gaps found.

### Blocking
*(none)*

### Should-fix (worth landing before tag, but I'm not blocking the merge on them)

**S1. Plotly axis text & donut bg-circle ignore `ThemeContext`**
`frontend/components/research/ResearchPriceChart.jsx` (PlotlyImpl `layout` at L168–L182) and `frontend/components/research/RegressionInsightCard.jsx` (donut bg-circle at L106 — `stroke="rgb(226 232 240)"`) hardcode light-theme values. Compare to `frontend/components/charts/RegressionChart.jsx:5,17,48` which pulls `useTheme()` and sets `font: { color: dark ? '#e2e8f0' : '#1e293b' }`. Tick labels and the donut background are barely visible in dark mode. Not a regression of any existing test — the e2e suite asserts on data-testids, not visual contrast — but follow-on polish work is warranted.

**S2. Axiom structured fields are message-body strings, not `extra={}` keys**
Every `logger.error` / `logger.exception` in `backend/app/routers/research.py` (L141, L188, L222, L293, L350, L402) passes `position_id` and `window` as positional `%s` args. The project's structured-logging convention (`backend/app/middleware.py:38`) uses `extra={"position_id": ..., "ticker": ...}` so Axiom can index those fields independently. Today the IDs are still searchable inside `message`, but won't be top-level columns in Axiom. NFR-6 partial. Quick win: switch to `extra={}` once the ticker is resolved from the position lookup.

### Nits

**N1.** `backend/app/services/research_price_history.py:78–104` duplicates `research_cache.py` (Worker A's helper) — `_payload_cache` + `_cache_get`/`_cache_set`/`clear_cache` are functionally identical to `research_cache.get/set/clear` with `ttl=_CACHE_TTL`. Worker B explicitly called this out as a PR-order workaround; now that both have landed, a one-pass dedup (delete 25 lines, swap calls) is mechanical. Tests still pass either way.

**N2.** `backend/app/services/research_regression.py:309` — `except (InvalidTickerError, DataFetchError, Exception) as exc:` — the tuple is redundant; bare `Exception` already covers the other two. Harmless, but a linter or reviewer will flag it again.

**N3.** `backend/app/services/research_regression.py` raises `RegressionSourceUnavailableError(f"Source unavailable for {ident}")` (L314) which then gets logged at the router. `ident` here is one of {ticker, "SPY", sector_etf, "DGS10"}, so the ticker effectively leaks into a log line — fine for ops, but worth noting the same logger.warning at L312 already does this without the f-string. Either path is consistent; the f-string is just slightly less symmetric with the other services.

**N4.** PR description says "Backend: 1438 passed, 13 skipped"; deviations call out 3 newly-skipped tests in `test_acceptance_phase4.py`. Worth confirming the rationale comment block (lines 192–214 in the file) survives a future cleanup pass — it's the only audit trail for why the old blanket yfinance ban was relaxed.

### Out-of-scope observations

- **yfinance transitive deps (peewee, protobuf, etc.)** — Worker A's report flagged them, and the new module imports yfinance lazily inside functions (`yfinance_client.py:78`) so cold start cost is contained. Docker image size will grow non-trivially; worth a follow-on ticket to evaluate `yfinance --no-deps` or a slimmer alternative if image size becomes a constraint.
- **5-min regression cache + 24h business/financials cache** — TTLs match plan §3.5 / NFR-7. The thesis endpoint correctly has no cache (cache-busting via the `version` counter is the intended pattern).
- **Render-phase hydration in `ThesisNoteEditor.jsx:88–93`** — correctly implements the React 19 + react-compiler "store-previous-prop" pattern. The identity-guard (`hydratedFrom !== data`) prevents the infinite-render trap.
- **Single-row version-counter for `position_notes`** matches plan §3.2; empty PUTs bump version (`research_thesis.py:120–124`), which is intentional per the spec.
- **`name` field on `BusinessResponse`** matches the PRD §"API Contracts → A" frozen contract. Worker A's deviation is the right call.
- **Sector ETF map alias** (`resolve_sector_etf = lookup_sector_etf`) handles both worker call-sites cleanly. Tests cover both names via the same module.

### Deviation roll-up (10 verified)

| # | Deviation | Verdict |
|---|---|---|
| 1 | research router dedup'd in `main.py` | ✓ registered once at L168 |
| 2 | `name` field on BusinessResponse | ✓ matches PRD |
| 3 | 3 acceptance_phase4 skips, narrower invariant elsewhere | ✓ AV client tests enforce "no yfinance for earnings" at L178/191/205 |
| 4 | inline cache in price_history vs research_cache helper | ✓ works; flagged as N1 dedup opportunity |
| 5 | trade-type normalizer + DEBUG-log on unknown | ✓ mapping covers all 7 DB values; `test_skips_unknown_trade_types` confirms intent |
| 6 | dual-name sector_etf_map | ✓ both names work, callers use correct name |
| 7 | 4 design-spec §4 ambiguities resolved | ✓ all 4 spec examples (SOFI/AAPL/TLT/biotech) covered by named tests |
| 8 | single-row version-counter for position_notes | ✓ UNIQUE on position_id, monotonic version, empty PUTs bump |
| 9 | single `useStockResearch` hook | ✓ Promise.allSettled fans the 5 sections out in parallel, no cross-section coupling |
| 10 | render-phase hydration in ThesisNoteEditor | ✓ correct React 19 pattern with identity guard |

### NFR roll-up

- NFR-1 (300ms skeleton) — ✓ `FullPageSkeleton` mounts immediately while position fetch is in-flight (`StockResearchPage.jsx:114`)
- NFR-2 (per-section parallel) — ✓ `Promise.allSettled(SECTIONS.map(fetchSection))` (`useStockResearch.js:130`)
- NFR-3 (per-section failure tile) — ✓ each card has its own `FailureTile` with onRetry; verified by `Section D 502 surfaces the section error tile while the rest render` test
- NFR-4 (no raw exception messages) — ✓ all 5xx paths use sanitized constants; `test_get_business_500_on_unexpected_exception` asserts the original exception text never leaks
- NFR-5 (thesis persistence) — ✓ 4000-char cap enforced both at Pydantic (`schemas.py:1217`) and router (`research_thesis.py:108`); GET returns empty-state shape on no row
- NFR-6 (Axiom log tagging) — partial; see S2
- NFR-7 (cache TTLs) — ✓ 24h business/financials, 15min price-history, 5min regression, no-cache thesis

Nice work pulling this off as a 5-worker parallel release. The contract freeze in the plan paid off — frozen test IDs end-to-end + per-worker file ownership disjoint by design meant the integration merge was 2 small conflicts (`research.py`, `alpha_vantage_client.py`) that resolved by union with no semantic loss.

File: backend/app/routers/research.py [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+            content={"detail": "Price source unavailable"},
+        )
+    except Exception as exc:  # noqa: BLE001 — generic 500 per CLAUDE.md
+        logger.error(
+            "Research price-history composition failed for position=%s "
+            "window=%s: %s",
+            position_id,
+            window,
+            exc,
+        )
```
[the remark this answers is not in the document]
Comment (ssandy33) [reply backend/app/routers/research.py]: Fixed in 0d4c9bc — logger.exception + extra={} per S2/CR #1

File: backend/app/services/alpha_vantage_client.py [hunk 2]
```diff
@@ -201,5 +202,418 @@ def get_cached_next_earnings_date(symbol: str) -> Optional[str]:
...
+            return cached
+
+    # Tier 2: SQLite durable cache
+    db_entry = _read_income_db_cache(symbol)
+    if db_entry:
+        cached, fetched_at = db_entry
+        if now - fetched_at < _CACHE_TTL:
+            _income_cache[symbol] = (cached, fetched_at)
+            return cached
+
```
[the remark this answers is not in the document]
Comment (ssandy33) [reply backend/app/services/alpha_vantage_client.py:327]: Fixed in 0d4c9bc — cache now stores full unsliced payload and slices on read; new regression test in test_alpha_vantage_client.py TestGetIncomeStatementCacheNPoisoning

File: backend/app/services/research_regression.py [hunk 3]
```diff
@@ -0,0 +1,463 @@
...
+
+# --- Public errors ---------------------------------------------------------
+
+
+class InsufficientRegressionDataError(Exception):
+    """Raised when fewer than ``MIN_OBSERVATIONS`` aligned trading days exist."""
+
+
+class RegressionSourceUnavailableError(Exception):
+    """Raised when an upstream data source (Schwab / FRED / sector ETF) fails."""
```
[the remark this answers is not in the document]
Comment (ssandy33) [reply backend/app/services/research_regression.py]: Fixed in 0d4c9bc — RegressionSourceUnavailableError now mirrors ResearchSourceUnavailable with a sanitized .detail attribute; raising sites no longer include the ticker in the message body. Router uses exc.detail for the 502 response.

File: backend/requirements.txt [hunk 4]
```diff
@@ -21,3 +21,4 @@ scipy==1.14.1
 PyJWT==2.12.0
 cryptography==46.0.7
 PyYAML==6.0.2
+yfinance>=0.2.50
```
[the remark this answers is not in the document]
Comment (ssandy33) [reply backend/requirements.txt]: Fixed in 0d4c9bc — pinned to yfinance==0.2.50

File: backend/tests/test_regression_summary.py [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+        factors=factors2, r_squared=0.70, sample_size=252
+    )
+    # market.contribution = 0.30, not strictly > 0.30, so "Diversified" wins.
+    assert out2.startswith("Diversified driver mix")
+    # First call's dominant is idiosyncratic at 0.30 (still tied with market);
+    # since max() takes the first occurrence of equal values among the
+    # iteration order, idiosyncratic wins here only if it appears first.
+    # We don't assert on `out`; the second case is the deterministic one.
+    del out
+
```
[the remark this answers is not in the document]
Comment (ssandy33) [reply backend/tests/test_regression_summary.py:171]: Fixed in 0d4c9bc — kept only the deterministic 0.30-boundary case as suggested

File: backend/tests/test_research_financials_router.py [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+    """Unknown ``position_id`` → 404 with sanitized detail."""
+    resp = client.get("/api/positions/does-not-exist/research/financials")
+    assert resp.status_code == 404
+    assert resp.json() == {"detail": "Position not found"}
+
+
+def test_get_financials_200_returns_av_payload(client):
+    """AV primary returns data → 200 with ``source=alphavantage``."""
+    pid = _create_position(client)
+    av_quarters = [_av_quarter(f"2025-0{i + 1}-30") for i in range(8)]
```
[the remark this answers is not in the document]
Comment (ssandy33) [reply backend/tests/test_research_financials_router.py]: Fixed in 0d4c9bc — fixture now uses real quarter-end dates (2025-03-31, 2024-12-31, ...)

File: backend/tests/test_sector_etf_map.py [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+def test_sector_etf_map_has_eleven_entries():
+    assert len(SECTOR_ETF_MAP) == 11
+
+
+def test_sector_etf_map_matches_frozen_contract():
+    """Every plan-§3.3 entry maps to the frozen ticker. No extra entries."""
+    assert SECTOR_ETF_MAP == _EXPECTED
+
+
+@pytest.mark.parametrize("sector,etf", list(_EXPECTED.items()))
```
[the remark this answers is not in the document]
Comment (ssandy33) [reply backend/tests/test_sector_etf_map.py]: Fixed in 0d4c9bc — both parametrize sites now use tuple argnames

File: frontend/components/research/ThesisNoteEditor.jsx [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+  const handleBlur = () => {
+    if (debounceTimer.current) {
+      window.clearTimeout(debounceTimer.current);
+      debounceTimer.current = null;
+    }
+    // Only flush if there's pending input vs the last-saved server state.
+    if (status === STATUS_SAVING || status === STATUS_IDLE) {
+      flush(body);
+    }
+  };
```
[the remark this answers is not in the document]
Comment (ssandy33) [reply frontend/components/research/ThesisNoteEditor.jsx:192]: Fixed in 0d4c9bc — handleBlur now compares body against lastSavedBodyRef.current and only flushes when dirty (ref is updated via useEffect to satisfy react-hooks/refs)

File: frontend/components/research/tradeMarkerVocabulary.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+/**
+ * Resolve a marker descriptor for a trade type, falling back to a
+ * conservative "unknown" pill so the chart never crashes on a new
+ * `trade_type` string the backend introduces in a future release.
+ */
+export function resolveTradeMarker(type) {
+  if (!type) return null;
+  return TRADE_MARKER_VOCAB[type] || null;
+}
```
[the remark this answers is not in the document]
Comment (ssandy33) [reply frontend/components/research/tradeMarkerVocabulary.js:35]: Fixed in 0d4c9bc — docstring updated to reflect the null-return contract (Option A). Callers are documented to handle null defensively.

File: frontend/hooks/useStockResearch.js [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+            name,
+            { data: null, loading: false, error: null },
+          ]),
+        ),
+      );
+      return;
+    }
+    // Position resolved — fan the five section fetches out in parallel.
+    await Promise.allSettled(SECTIONS.map((name) => fetchSection(name)));
+  }, [positionId, fetchSection]);
```
[the remark this answers is not in the document]
Comment (ssandy33) [reply frontend/hooks/useStockResearch.js:150]: Fixed in 0d4c9bc — added requestSeqRef monotonic token, every setState in fetchSection/refetchAll/saveThesis is now guarded so a stale request cannot clobber fresh state

File: backend/app/services/research_financials.py [hunk 11]
```diff
@@ -0,0 +1,282 @@
...
+            db.add(AppSetting(key=key, value=value))
+        else:
+            entry.value = value
+        db.commit()
+    except (SQLAlchemyError, ValueError) as exc:
+        logger.debug("Failed to write %s to app_settings: %s", key, exc)
+        try:
+            db.rollback()
+        except SQLAlchemyError:
+            pass
```
[the remark this answers is not in the document]
Comment (ssandy33) [reply backend/app/services/research_financials.py:209]: Deferred to follow-up #284 — tracked separately. Reason: the duplication is intentional during v1.1.0 worker fan-out and is well-documented in both file headers; consolidation is pure tech-debt with no user-visible impact and warrants its own focused PR.

File: frontend/components/research/RegressionInsightCard.jsx [hunk 12]
```diff
@@ -0,0 +1,355 @@
...
+    return { key: 'market', label: name, displayName: 'Market' };
+  }
+  if (name === 'DGS10' || name.endsWith('YIELD')) {
+    return { key: 'rates', label: name, displayName: 'Rates' };
+  }
+  if (name.startsWith('XL') || factor?.role === 'sector') {
+    return { key: 'sector', label: name, displayName: 'Sector' };
+  }
+  return { key: 'sector', label: name, displayName: 'Sector' };
+}
```
[the remark this answers is not in the document]
Comment (ssandy33) [reply frontend/components/research/RegressionInsightCard.jsx:211]: Deferred to follow-up #285 — tracked separately. Reason: the v1.1.0 backend only emits four canonical factor names (market/sector/rates/idiosyncratic) and the categorizer matches all four explicitly; the fallback branch is only reachable if the backend vocabulary expands, which is a forward-defense polish pass not blocking the v1.1.0 release.
